### PR TITLE
core: remove wildcard permission in rbac

### PR DESCRIPTION
**Pull request by parth-gr**
Reduce the RBAC scope to the minimum necessary
permissions for rook to operator

Signed-off-by: parth-gr <paarora@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/monitoring/rbac.yaml [hunk 1]
```diff
@@ -7,12 +7,32 @@ metadata:
...
-- apiGroups:
-  - monitoring.coreos.com
-  resources:
-  - '*'
-  verbs:
-  - '*'
+  - apiGroups:
+      - monitoring.coreos.com
+    resources:
+      - cephclients
```
Comment (travisn) [inline deploy/examples/monitoring/rbac.yaml:13]: We don't need all of the rook CRDs in this list, they don't apply to the monitoring.coreos.com api group. I believe we would only need `servicemonitors` in this list. Another possibility is `promeusrules`, but the rook operator doesn't create those anymore so it shouldn't need access to that.

**Review by travisn — CHANGES_REQUESTED** [review]

File: deploy/examples/monitoring/rbac.yaml (hunk 1 above)
Comment (parth-gr) [reply deploy/examples/monitoring/rbac.yaml]: Yaa that's right,
I just didn't saw that its for monitor

Thanks

**Review by travisn — COMMENTED** [review]
The changes look good. I believe this rbac is not tested in the CI so we need manual verification to make sure the operator doesn't have any errors in the log. If you create this yaml then set `monitoring: true` in the cluster CR, it should be sufficient to trigger, and possibly restart the operator. Also perhaps see the [prometheus config](https://rook.io/docs/rook/latest/Storage-Configuration/Monitoring/ceph-monitoring/#prometheus-alerts).

File: deploy/examples/monitoring/rbac.yaml [hunk 2]
```diff
@@ -27,9 +32,9 @@ roleRef:
   kind: Role
   name: rook-ceph-monitor
 subjects:
-- kind: ServiceAccount
-  name: rook-ceph-system
-  namespace: rook-ceph
+  - kind: ServiceAccount
```
Comment (sp98) [inline deploy/examples/monitoring/rbac.yaml:35]: Can you check the indentation again? Seems to be a bit off as an extra space is added in all the changes.

File: deploy/examples/monitoring/rbac.yaml [hunk 3]
```diff
@@ -7,12 +7,17 @@ metadata:
...
-  verbs:
-  - '*'
+  - apiGroups:
+      - monitoring.coreos.com
+    resources:
+      - servicemonitors
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (sp98) [inline deploy/examples/monitoring/rbac.yaml:17]: `Watch` and `Delete` should not be needed.

File: deploy/examples/monitoring/rbac.yaml (hunk 2 above)
Comment (parth-gr) [reply deploy/examples/monitoring/rbac.yaml:35]: Actually a go indent is added or been updated,

So if you touch any of the file that is old format will change

File: deploy/examples/monitoring/rbac.yaml (hunk 3 above)
Comment (parth-gr) [reply deploy/examples/monitoring/rbac.yaml:17]: For read access we need `watch` also https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-example
`Delete` access of service monitor I am not sure

Comment (travisn) [reply deploy/examples/monitoring/rbac.yaml:17]: We may need `patch`, but perhaps the `update` already covers updates during reconcile. I don't recall the difference.

**Review by travisn — APPROVED** [review]
As discussed, let's go with this change, and no need for delete or patch, but we will let it stabilize in master for now.

**Pull request merged**